packages/slate-editor/src/extensions/variables/VariableMenu.tsx

```diff
@@ -36,6 +36,11 @@ export function VariableMenu({ container, element, onClose, variables }: Props)
 
     function handleChangeType(newType: string) {
         updateVariable(editor, { key: newType });
+
+        const newOption = options.find(({ value }) => value === newType);
+        if (newOption && !newOption.withFallback) {
+            updateVariable(editor, { fallback: '' });
+        }
```

Comment:
This can be combined into a single `updateVariable()` call:

```ts
if (newOption && !newOption.withFallback) {
  updateVariable(editor, { key: newKey, fallback: '' });
} else {
  updateVariable(editor, { key: newKey });
}
```